src/dataclass_binder/_impl.py

```diff
@@ -314,6 +325,15 @@ def _bind_to_single_type(self, value: object, field_type: type, context: str) ->
                 if not isinstance(value, str):
                     raise TypeError(f"Expected TOML string for path '{context}', got '{type(value).__name__}'")
                 return field_type(value)
+            elif issubclass(field_type, ReprEnum):
+                return field_type(value)
+            elif issubclass(field_type, Enum):
+                if not isinstance(value, str):
+                    raise TypeError(f"'{value}' is not a valid key for enum '{field_type}', must be of type str")
+                for enum_value in field_type:
+                    if enum_value.name.lower() == value.lower():
```

Comment:
Small optimization: you could store `value.lower()` in a local variable, to avoid converting the same string multiple times.

Reply:
Since there is no other case where this is used in this function I'm not sure this would help :thinking: